db/model/Gdoc/GdocFactory.ts

```diff
@@ -309,7 +309,8 @@ export async function createOrLoadGdocById(
 export async function loadGdocFromGdocBase(
     knex: KnexReadWriteTransaction,
     base: OwidGdocBaseInterface,
-    contentSource?: GdocsContentSource
+    contentSource?: GdocsContentSource,
+    loadLatestDataInsights?: boolean
```

Comment:
This is ugly, but I didn't find a better way to prevent infinite loop when loading data insights, which in turn have `latestDataInsights`.

Reply:
Loading images depends on loading linked documents, and I thought there might be also other hidden dependencies. So it seemed not worth it to create a special case for this use case that we'll need to maintain. Do you think I should give the special case a go despite this?

Single object param - it would be inconsistent with similar other functions in the module, and changing all of them seems like too big change for little benefit. I think it makes sense if it would make using the function significantly easier (i.e. programmatically constructing the args). Otherwise, I don't think it's worth it for aesthetic reasons only (I might be missing how it helps with grep-ability).

Reply:
Loading linked documents is only necessary to get their featured images, in case we're using a prominent link (which uses the featured image as a thumbnail)

So I think for DIs where we're very likely not going to have a prominent link to another post in the first paragraph, and where we have a fallback for when no featured image is present, it's completely safe.

For the param thing, my personal rule is to switch after 3+ arguments, but it's just a personal preference that I picked up from [clean code](https://github.com/ryanmcdermott/clean-code-javascript?tab=readme-ov-file#function-arguments-2-or-fewer-ideally) - no worries if you prefer consistency.